lightkurve/tests/test_lightcurve.py

```diff
@@ -360,3 +360,19 @@ def test_flatten():
     assert(len(flat_lc.time) == 5)
     assert(np.isfinite(flat_lc.flux).sum() == 3)
     assert(np.isfinite(flat_lc.flux_err).sum() == 3)
+
+
+def test_flatten_robustness():
+    """Test various special cases for flatten()."""
+    # flatten should work with integer fluxes
+    lc = LightCurve([1, 2, 3, 4, 5, 6], [10, 20, 30, 40, 50, 60])
+    flat_lc = lc.flatten(window_length=3, polyorder=1)
```

Comment:
This line raises an exception if we use `flatten_lc.flux /= trend_signal` in the implementation as opposed to `flatten_lc.flux = flatten_lc.flux / trend_signal`.